### PR TITLE
Move kind bootstrap into deployer

### PR DESCRIPTION
**Pull request by pebrc**
Kind is currently a bit of a snowflake in our e2e test setup. This PR aims at moving it under the deployer umbrella. The motivation is mainly to have a uniform way of installing manifests (with Kubernetes version awareness) for https://github.com/elastic/cloud-on-k8s/pull/4489 

This PR drops the image loading functionality we had in the stand-alone version. I believe we don't need it anymore given that our images are all public and at least in my tests kind was able to pull them. I am however happy to reconsider if there are concerns around this removal. 

I have run a smoke test on a local Jenkins instance which went OK.

### CODE REVIEW

File: hack/deployer/runner/kind.go [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+		"ClusterName":     k.plan.ClusterName,
+		"Args":            args,
+	}
+	// We are mounting the shared volume into the installer container and configure it to be the HOME directory
+	// this is mainly so that the GCloud tooling picks up the authentication information correctly as the base image is
+	// scratch+curl and thus an empty
+	// We are mounting tmp as the installer needs a scratch space and writing into the container won't work
+	cmd := NewCommand(`docker run --rm \
+		-v {{.SharedVolume}}:/home \
+        -v /var/run/docker.sock:/var/run/docker.sock \
```
Comment (thbkrkr) [inline hack/deployer/runner/kind.go:200]: Nit: convert spaces to tabs.

File: hack/deployer/runner/kind.go [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+	params := map[string]interface{}{
+		"SharedVolume":    SharedVolumeName(),
+		"KindClientImage": k.clientImage,
+		"ClusterName":     k.plan.ClusterName,
+		"Args":            args,
+	}
+	// We are mounting the shared volume into the installer container and configure it to be the HOME directory
+	// this is mainly so that the GCloud tooling picks up the authentication information correctly as the base image is
+	// scratch+curl and thus an empty
+	// We are mounting tmp as the installer needs a scratch space and writing into the container won't work
```
Comment (thbkrkr) [inline hack/deployer/runner/kind.go:197]: I don't see this mount.

Comment (pebrc) [reply hack/deployer/runner/kind.go:197]: Sloppy copy pasta from OCP :-(

File: .ci/setenvconfig [hunk 3]
```diff
@@ -127,6 +124,19 @@ CLUSTER_NAME=kind-$BUILD_NUMBER
...
+overrides:
+  clusterName: eck-e2e-$BUILD_NUMBER
+  clientVersion: $3
+  vaultInfo:
+    address: $VAULT_ADDR
+    roleId: $VAULT_ROLE_ID
+    secretId: $VAULT_SECRET_ID
+  kind:
+    nodeImage: $2
+    ipFamily: $4
```
Comment (thbkrkr) [inline .ci/setenvconfig:138]: Nit: keep the variables indirection to clearly explicit the arguments required to call `e2e/kind-k8s-version`.
```sh
  e2e/kind-k8s-versions)
######################################

kindVersion=$2
kindNodeImage=$3
ipFamily=$4

write_env <<ENV
...
E2E_TEST_ENV_TAGS="kind,$ipFamily"   // use
ENV
write_deployer_config <<CFG
id: kind-ci
overrides:
  clusterName: eck-e2e-$BUILD_NUMBER
  clientVersion: ${kindVersion}
  vaultInfo:
    address: $VAULT_ADDR
    roleId: $VAULT_ROLE_ID
    secretId: $VAULT_SECRET_ID
  kind:
    nodeImage: ${kindNodeImage}
    ipFamily: ${ipFamily}
CFG
;;
```

File: hack/deployer/runner/settings.go [hunk 4]
```diff
@@ -104,6 +105,13 @@ type EKSSettings struct {
 	WorkDir   string `yaml:"workDir"`
 }
 
+type KindSettings struct {
+	NodeCount    int      `yaml:"nodeCount"`
+	NodeImage    string   `yaml:"nodeImage"`
+	ImagesToLoad []string `yaml:"imagesToLoad"`
```
Comment (thbkrkr) [inline hack/deployer/runner/settings.go:111]: To be deleted because no longer used?

**Review by thbkrkr — APPROVED** [review]
Nice. LGTM. Just left a few nits.

It works well in local (`make switch-kind bootstrap-k8s` took 1m33, deployer-kind client and kind images already pulled).

**Pull request merged**